src/collectives_c.c4

```diff
@@ -150,10 +227,33 @@ shmem_sync(int PE_start, int logPE_stride, int PE_size, long *pSync)
         SHMEM_ERR_CHECK_SYMMETRIC(pSync, sizeof(long)*SHMEM_REDUCE_SYNC_SIZE); \
                                                                         \
         shmem_internal_op_to_all(target, source, nreduce, sizeof(TYPE), \
-                                 PE_start, logPE_stride, PE_size,       \
+                                 PE_start, 1 << logPE_stride, PE_size,  \
                                  pWrk, pSync, IOP, ITYPE);              \
     }
 
+#define SHMEM_DEF_REDUCE(STYPE,TYPE,ITYPE,SOP,IOP)                      \
+    int SHMEM_FUNCTION_ATTRIBUTES                                       \
+    shmemx_##STYPE##_##SOP##_reduce(shmemx_team_t team, TYPE *dest,     \
+                                       const TYPE *source,              \
+                                       size_t nreduce)                  \
+    {                                                                   \
+        SHMEM_ERR_CHECK_INITIALIZED();                                  \
+        SHMEM_ERR_CHECK_TEAM_VALID(team);                               \
+        SHMEM_ERR_CHECK_NON_NEGATIVE(nreduce);                          \
```

Comment:
`size_t` is unsigned, so this check is not needed (and should also generate a compile time warning about signed/unsigned comparison).

Reply:
Strange... I don't see a warning with clang + `-Wall -Wextra`.  Good catch!